### DISCUSSION TITLE
Add descriptions for plugins?

### DISCUSSION THREAD

**Discussion by RobJacobson** [ohmyzsh/ohmyzsh#11650] [Ideas]
I'm a newbie to OMZ. I find your list of plugins to be really interesting, but also rather intimidating. I have no idea of what each one does or which ones I should install. I could click through each link, of course, but with ~100 different options, that's not realistic. I've come across this page several times, and each time I conclude, "this is too complicated" and skip installing plugins.

Some of the underlying packages are self-evident, like "Python." However, most of these names mean nothing to me. E.g., what does "autojump" do? 

Even the description after I click through is rather bare: "This plugin loads the [autojump navigation tool](https://github.com/wting/autojump)." This sounds intriguing, but it isn't informative unless (a) you already know what autojump does or (b) you have the time to click through to the underlying package and read the detailed description.

Would it be possible to expand this page with a description for each plugin and the underlying package? The descriptions would ideally have a one or two sentence summary about "why should I care?" -- or "what makes this awesome?" I.e., more than "this plugin loads X" or "this plugin adds syntax highlighting for Y." Rather, there would ideally be a short summary of the underlying package, like "Autojump is X and does Y." Perhaps a table format would work better, not a list.

I'd be happy to volunteer to write a first pass and submit a pull request, although it would require some research and time. I wouldn't want to do this unless there was buy-in. It would help if anyone could volunteer to review a first draft before I submit a formal pull request.

Just my two cents. Thanks for maintaining a great package!

PS: Is there a correct pronunciation for "ohmyzsh"? In my head, it's "oh my zosh" (like the English name"Josh"), but I realized that "zash," "zesh," "zish" or "zush" are also possibilities. The last two are hard to pronounce.  :-)

**Comment by carlosala** [comment]
It's a great idea! I see this especially for plugins like `autojump`, `z`, etc. Those are tools that you'll not be using outside here, you'll find them when digging into shell.
For `rails` plugin, for instance, it's not needed, as you'll only enable it if you use `rails`.

**Reply by mcornella** [reply]
Created https://github.com/ohmyzsh/wiki/issues/59 to track this.

**Comment by barryam3** [comment]
> PS: Is there a correct pronunciation for "ohmyzsh"? In my head, it's "oh my zosh" (like the English name"Josh"), but I realized that "zash," "zesh," "zish" or "zush" are also possibilities. The last two are hard to pronounce. :-)

Apparently "zsh" is pronounced "z-shell."

https://refine.dev/blog/zsh-vs-bash/#what-is-zsh

The creator of Oh My Zsh says it the same way when saying the framework's name: "Oh My Z-shell".

https://www.youtube.com/watch?v=jmDiZbfX8Ug

I assumed it was intended to rhyme with "oh my gosh" until I heard a coworker say "oh my zeesh," which led me down this rabbit hole. There you go.

**Comment by mcornella** [comment]
Descriptions are now added to the [wiki](https://github.com/ohmyzsh/ohmyzsh/wiki/Plugins). Click on the link to the README to learn more about the plugin.
